index.js

```diff
@@ -3,14 +3,16 @@
 const fp = require('fastify-plugin')
 const nano = require('nano')
 
-async function fastifyCouchDB (fastify, options, next) {
-  if (fastify.couch) {
+function fastifyCouchDB (fastify, options, next) {
+  if (fastify.hasDecorator('couch')) {
     next(new Error('fastify-couchdb has already registered'))
   } else {
-    fastify.couch = await nano(options)
-  }
+    const couch = nano(options)
+    fastify.decorate('couch', couch)
+    fastify.decorateRequest('db', couch)
```

Comment:
This is not really needed.

Reply:
Ahh, I think I understand. So does this mean that `fastify.decorateRequest('db', couch)` can be removed then as we can access `couch` via `this` in the routes?

Reply:
Yep!